src/assets/data/markdown/ionic-react/style-the-app.md

```diff
@@ -0,0 +1,306 @@
+  text-align: center;
+}
+```
+
+We will expand this shortly to contain more styles. As the styles grow, we will want to look at splitting them out into seperate files that are imported here, but for now we can just add our styles to the `global.css` file.
+
+### Shadow DOM
+
+Most of the components in the Ionic Framework use <a href="https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM" target="_blank">Shadow DOM</a>. This works very similar to the `scoped` discussion above (just without the extra classes), allowing the framework developers to apply styling to the individual components without having that styling bleed out into other components. This also you as a developer to use the component without having to worry about the details of the component's construction when styling your own application.
+
+However, it also means that different techniques must be employed when styling these components. The authors of the components must provide you with a styling API that defines the ways in which they intend for you to be able to style these components. The two mechanisms that make this possible are `CSS Custom Properties` (also called `CSS Variables`) and `Shadow Parts`.
+
+### Light DOM
+
+There really is no official term called "Light DOM" but it is a term that is common referred to items outside of a shadow root that you can style via traditional means. Even for many components that use shadow DOM you are still able to apply "light DOM" styling to them. For example:
+
+```css
+ion-card {
+  border-radius: 20px;
+}
+```
+
+The `ion-card` uses shadow DOM, but that is for the internal structure. The border radius applies to the card itself, which is visible in the normal DOM tree, and thus can be styled via traditional means.
+
+### CSS Custom Properties
+
+The Ionic Framework makes heavy use of CSS Custom Properties. They are used to define the color scheme, default padding, etc. They are also used heavily in order to style individual components. Have a look at the <a href="https://ionicframework.com/docs/api/button#css-custom-properties">CSS Custom Properties for a button</a> as an example. This defines an API for the ways in which the framework authors intend for you to be able to style a button. You can do so as such:
+
+```css
+ion-button {
+  --opacity: 0.75;
+}
+```
+
+This will apply a 75% opacity to all buttons in the application. You probably want to be a bit more restrictive with your selector, but that is the basic idea with the custom properties.
+
+### Shadow Parts
+
+A newer specification is <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/::part">Shadow Parts</a>. This allows the component author to tag specific parts of the component as something you can apply styles to as if it were in the light DOM. This allows component developers to still restrict how a component can be styling without having to redefine all of the ways an underlying element can be styled via a CSS Custom Property.
```

Comment:
Typo: `how a component can be styling` should be `how a component can be styled`.